runtime-modules/content/src/permissions/curator_group.rs

```diff
@@ -1,4 +1,48 @@
 use super::*;
+use sp_std::collections::btree_map::BTreeMap;
+#[cfg(feature = "std")]
+use strum_macros::EnumIter;
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub struct ContentModerationPermissions {
```

Comment:
Would it perhaps be a more efficient representation to just have something like `BTreeSet<ContentModerationAction>`, where presence `a` <=> allowed to do action `a`. More importantly than the storage efficiency in storage or in extrinsics is that you get rid of the conversion layer you need in places like `can_perform_action`.

Reply:
I was considering that, but I noticed that I'm unable to use nested `BTreeSet` (ie. `BTreeMap<T, BTreeSet<T>>`) as an extrinsic argument, so I needed some other type to represent the permissions. Alternatively I could use `Vec`, but then I'd need to introduce additional validation to make sure each `ContentModerationAction` only existis there once, so I went for `Struct` instead.

Reply:
This turned out to be caused by missing `PartialOrd, Ord` traits for `ContentModerationAction`. After adding those I was able to succesfully switch to the nested `BTreeSet<ContentModerationAction>` approach. Done in https://github.com/Joystream/joystream/pull/3366/commits/001745b49cb9fd57fd3c41646c95e172635ae6e5